## Commands w () and w/o ()

In [3]:
import pandas as pd
movies = pd.read_csv('http://bit.ly/imdbratings')

In [4]:
movies.head()


star_rating                     title content_rating   genre  duration  \
0          9.3  The Shawshank Redemption              R   Crime       142   
1          9.2             The Godfather              R   Crime       175   
2          9.1    The Godfather: Part II              R   Crime       200   
3          9.0           The Dark Knight          PG-13  Action       152   
4          8.9              Pulp Fiction              R   Crime       154   

                                         actors_list  
0  [u'Tim Robbins', u'Morgan Freeman', u'Bob Gunt...  
1    [u'Marlon Brando', u'Al Pacino', u'James Caan']  
2  [u'Al Pacino', u'Robert De Niro', u'Robert Duv...  
3  [u'Christian Bale', u'Heath Ledger', u'Aaron E...  
4  [u'John Travolta', u'Uma Thurman', u'Samuel L....

In [5]:
movies.describe()

star_rating    duration
count   979.000000  979.000000
mean      7.889785  120.979571
std       0.336069   26.218010
min       7.400000   64.000000
25%       7.600000  102.000000
50%       7.800000  117.000000
75%       8.100000  134.000000
max       9.300000  242.000000

In [6]:
movies.shape


(979, 6)

In [8]:
type(movies)

pandas.core.frame.DataFrame

979 rows 6 col

methods have () action oriented verbs
atribues have no () and they are descriptions of who you are (age height)
functions start with pd.


In [10]:
movies.describe(include = ['object'])

title content_rating  genre  \
count       979            976    979   
unique      975             12     16   
top     Dracula              R  Drama   
freq          2            460    278   

                                              actors_list  
count                                                 979  
unique                                                969  
top     [u'Daniel Radcliffe', u'Emma Watson', u'Rupert...  
freq                                                    6

In [12]:
movies.describe(include = 'all')

star_rating    title content_rating  genre    duration  \
count    979.000000      979            976    979  979.000000   
unique          NaN      975             12     16         NaN   
top             NaN  Dracula              R  Drama         NaN   
freq            NaN        2            460    278         NaN   
mean       7.889785      NaN            NaN    NaN  120.979571   
std        0.336069      NaN            NaN    NaN   26.218010   
min        7.400000      NaN            NaN    NaN   64.000000   
25%        7.600000      NaN            NaN    NaN  102.000000   
50%        7.800000      NaN            NaN    NaN  117.000000   
75%        8.100000      NaN            NaN    NaN  134.000000   
max        9.300000      NaN            NaN    NaN  242.000000   

                                              actors_list  
count                                                 979  
unique                                                969  
top     [u'Daniel Radcliffe', u'Emma Watson', u'Rupert...  
freq                                                    6  
mean                                                  NaN  
std                                                   NaN  
min                                                   NaN  
25%                                                   NaN  
50%                                                   NaN  
75%                                                   NaN  
max                                                   NaN